dropshot_endpoint/src/params.rs

```diff
@@ -0,0 +1,302 @@
+// Copyright 2024 Oxide Computer Company
```

Comment:
post-accept, moved param-related code into its own file because it was getting quite big and making `endpoint.rs` harder to read. This code is also going to be mostly shared with channels in the future.